### PR TITLE
Feature/vessels ui

### PR DESCRIPTION
**Pull request by Swapna-Sahu**
Created the vessels ui component which is  working in storybook.

### CODE REVIEW

File: src/client/components/Vessels-UI/VesselsUI.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    state = {
+      loading:true
+    }
+
+    /*componentDidMount() {
+        const url = "https://localhost:5000/api/vessels/";
+        const response = async fetch(url);
+        const data = await response.json();
+        console.log("inside ComponentDidMount",data);
+    }*/
```
Comment (zkwsk) [inline src/client/components/Vessels-UI/VesselsUI.js]: Please do not commit commented out code. If you want to save it for later, make a new branch from your feature branch`checkout -b [future-branch]`, then go back to your feature branch and delete the changes again before pushing.

**Review by zkwsk — CHANGES_REQUESTED** [review]
I just had a look in storybook and looks really good, but I have some requests.

First, I would like for the category (storiesOf) in Storybook to be "Vessels".

Then, if you look at the task again, you're actually supposed to make a list of vessels. I only see one. And since this view is not revealing all details about the vessel (unlike vessel details) I think we should call this one "vessel teaser".

So I would like to see two compoents in Storybook. One component is called "Vessel List" and the other one is called "Vessel Teaser".

"Vessel List" does not really have a UI (yet), it's just mapping over all of the "Vessel Teasers" you might feed into it. In the future you might put something like a search field or filter controls on the "Vessel List" and it is also the component that's most likely to be stateful (or maybe you need to move state higher up - I don't know yet).

In Storybook under the category "Vessels" I would like to see both the list component with teaser components inside and the teaser component under its own individual story. So if someone else maybe need to build a different kind of list, he can see how the teaser works on it's own.

Since you're already using knobs 👏 I suggest you look into using the object knob on the "Vessel List" to have an array of "Vessel Teasers" similar to what Niels did with Waypoints in his component.

**Comment by Swapna-Sahu** [comment]
1. Renamed the category of storiesOf Vessels.
2. Created two components VesselsList and VesselTeaser
3. VesselsList is only mapping VesselTeaser
4. knobs is showing all vessels

File: src/client/components/Vessels-UI/VesselsUI.js (hunk 1 above)
Comment (Swapna-Sahu) [reply src/client/components/Vessels-UI/VesselsUI.js]: Removed the file because it is not required.

File: src/client/stories/index.js [hunk 2]
```diff
@@ -3,10 +3,17 @@ import { storiesOf } from "@storybook/react";
...
 storiesOf("Button", module)
   .add("with text", WithTextStory)
   .add("with emoji", WithEmojiStory);
 
 const MapStories = storiesOf("Map", module).add("MapComponent", MapStory);
 MapStories.addDecorator(withKnobs);
+
+const VesselsStories = storiesOf("Vessels", module).add(
+  "VesselsComponent",
+  VesselsStory
```
Comment (zkwsk) [inline src/client/stories/index.js]: You need to have the components separately in Storybook:
 - VesselTeaser
 - VesselList

File: src/client/components/VesselTeaser.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from "react";
+
+function VesselsComponent({
```
Comment (zkwsk) [inline src/client/components/VesselTeaser.js]: VesselTeaser

File: src/client/components/VesselsList.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from "react";
+import Vessels from "./VesselTeaser";
+
+function VesselsList({ vessels = [] }) {
+  const vesselsList = vessels.map(
+    ({ id, vesselName, departureFrom, arrivesAt, etd, eta }) => (
+      <Vessels
```
Comment (zkwsk) [inline src/client/components/VesselsList.js:7]: VesselTeaser

**Review by zkwsk — CHANGES_REQUESTED** [review]
Looks great. Just a few minor fixes.

File: src/client/stories/index.js (hunk 2 above)
Comment (Swapna-Sahu) [reply src/client/stories/index.js]: Created two separate storybook
VesselTeaser
VesselList

File: src/client/components/VesselsList.js (hunk 4 above)
Comment (Swapna-Sahu) [reply src/client/components/VesselsList.js:7]: Resolved

File: src/client/components/VesselTeaser.js (hunk 3 above)
Comment (Swapna-Sahu) [reply src/client/components/VesselTeaser.js]: Vessel

File: src/client/stories/index.js [hunk 5]
```diff
@@ -3,10 +3,24 @@ import { storiesOf } from "@storybook/react";
...
+import { VesselTeaserStory } from "./VesselTeaser";
 
 storiesOf("Button", module)
   .add("with text", WithTextStory)
   .add("with emoji", WithEmojiStory);
 
 const MapStories = storiesOf("Map", module).add("MapComponent", MapStory);
 MapStories.addDecorator(withKnobs);
+
+const VesselsListStories = storiesOf("VesselList", module).add(
```
Comment (zkwsk) [inline src/client/stories/index.js:16]: Rename the category (storiesOf) to `Vessels` and add both `VesselTeaser` and `VesselList` under the same category.

File: src/client/components/VesselsList.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from "react";
+import Vessels from "./VesselTeaser";
```
Comment (zkwsk) [inline src/client/components/VesselsList.js:2]: import it under the name `VesselTeaser` for consistency.

File: src/client/components/VesselTeaser.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import React, { Component } from "react";
+
+function Vessel({ id, vesselName, departureFrom, arrivesAt, etd, eta }) {
```
Comment (zkwsk) [inline src/client/components/VesselTeaser.js:3]: Rename to `VesselTeaser`. Check to make sure that's consistent throughout the PR.

File: src/server/api/routes/user.router.js [hunk 8]
```diff
@@ -14,6 +14,7 @@ const ROLES = require("../../constants/roles");
 const userController = require("../controllers/user.controller");
 
 // ENDPOINT: /api/users/ :GET
+
 router.get("/", authorizeUser(ROLES.SUPER_USER), (req, res, next) => {
```
Comment (zkwsk) [inline src/server/api/routes/user.router.js:18]: Did you intend to add these empty lines in a couple of files that are unrelated to your PR? ;)

**Review by zkwsk — CHANGES_REQUESTED** [review]

File: src/client/stories/index.js (hunk 5 above)
Comment (Swapna-Sahu) [reply src/client/stories/index.js:16]: Renamed the category (storiesOf) to Vessels and add both VesselTeaser and VesselList under the same category.

File: src/client/components/VesselTeaser.js (hunk 7 above)
Comment (Swapna-Sahu) [reply src/client/components/VesselTeaser.js:3]: renamed to VesselTeaser

File: src/server/api/routes/user.router.js (hunk 8 above)
Comment (Swapna-Sahu) [reply src/server/api/routes/user.router.js:18]: Do I need to delete the empty line?

File: src/client/components/VesselsList.js (hunk 6 above)
Comment (Swapna-Sahu) [reply src/client/components/VesselsList.js:2]: renamed VesselTeaser

**Comment by Swapna-Sahu** [comment]
https://github.com/HackYourFuture-CPH/ov-class09-fp/pull/88
